modules/cloud-function/variables.tf

```diff
@@ -136,7 +136,24 @@ variable "trigger_config" {
   type = object({
     event    = string
     resource = string
-    retry    = bool
+    retry    = optional(bool)
+  })
+  default = null
+}
+
+variable "trigger_config_v2" {
```

Comment:
Would it make sense to have `trigger_config` be an object with `v1` and `v2` members? I'm not sure, just asking how you feel about it.

Reply:
I did not that originally do that to preserve backward compatibility. But having it within one variable allows setting condition that checks that only one was provided, which I consider an advantage.

Pushed changes containing this approach. The only thing I don't like about it is the need to use `try(...)` to check existence of fields and this prevents detecting typos in these constructs.